cluster-operator/src/main/java/io/strimzi/operator/cluster/model/ModelUtils.java

```diff
@@ -509,6 +510,21 @@ public static void setClusterOperatorNetworkPolicyNamespaceSelector(NetworkPolic
                 peer.setNamespaceSelector(new LabelSelector());
             }
         }
+    }
 
+    /**
+     * Checks if the section of the custom resource has any metrics configuration and sets it in the AbstractModel.
+     *
+     * @param model                     The cluster model where the metrics will be configured
+     * @param resourceWithMetrics       The section of the resource with metrics configuration
+     */
+    public static void parseMetrics(AbstractModel model, HasConfigurableMetrics resourceWithMetrics)   {
+        if (resourceWithMetrics.getMetrics() != null) {
+            model.setMetricsEnabled(true);
+            model.setMetricsConfig(resourceWithMetrics.getMetrics().entrySet());
+        } else if (resourceWithMetrics.getMetricsConfig() != null)    {
+            model.setMetricsEnabled(true);
+            model.setMetricsConfigInCm(resourceWithMetrics.getMetricsConfig());
+        }
```

Comment:
If the user, by mistake, specifies both `metrics` and `metricsConfig` fields in the `Kafka` resource we are giving precedence to the "deprecated/old" way. Is this on purpose for backward compatibility or we should use the "if" in the other way around?

Reply:
Indeed ... it probably makes sense to use the new first. Thanks